packages/core/src/unit-of-work/UnitOfWork.ts

```diff
@@ -473,8 +473,8 @@ export class UnitOfWork {
   private unwrapReference<T extends AnyEntity<T>, U extends AnyEntity | Reference<T> | Collection<AnyEntity> | Primary<T> | (AnyEntity | Primary<T>)[]>(entity: T, prop: EntityProperty<T>): U {
     const reference = entity[prop.name] as U;
 
-    if (reference instanceof Reference) {
-      return reference.unwrap();
+    if (Reference.isReference(reference)) {
+      return reference.unwrap() as any as U;
```

Comment:
same here, as we are still referencing the `Reference` class

```suggestion
      return reference.unwrap() as unknown as U;
```

Reply:
That seems to break the TypeScript build
```
src/entity/Reference.ts(34,7): error TS2322: Type '(T & Reference<AnyEntity<unknown, never>>) | ({ [K in PK]: T[K]; } & Reference<T> & Reference<AnyEntity<unknown, never>>)' is not assignable to type 'IdentifiedReference<T, PK>'.
  Type 'T & Reference<AnyEntity<unknown, never>>' is not assignable to type 'IdentifiedReference<T, PK>'.
    Type 'T & Reference<AnyEntity<unknown, never>>' is not assignable to type 'Reference<T>'.
      Types of property '__helper' are incompatible.
        Type 'IWrappedEntityInternal<AnyEntity<unknown, never>, unique symbol> | undefined' is not assignable to type 'IWrappedEntityInternal<T, keyof T> | undefined'.
          Type 'IWrappedEntityInternal<AnyEntity<unknown, never>, unique symbol>' is not assignable to type 'IWrappedEntityInternal<T, keyof T>'.
            The types of '__meta.properties' are incompatible between these types.
              Type '{}' is not assignable to type '{ [K in keyof T & string]: EntityProperty<T>; }'.
```

Reply:
uh what? casting as `unknown` breaks the build? the error is from different line than this

Reply:
its weird that we now get this error, but it should not be connected to this change, its about `Reference.create()` method

i guess we need to do this to fix it:

```typescript
if (entity instanceof Reference) {
  return entity as IdentifiedReference<T, PK>; // or maybe even `as unknown` first
}
```

Reply:
Yup, that fixes it, thanks!